virttest/storage.py

```diff
@@ -379,6 +390,17 @@ def auth_info_define_by_params(cls, image, params):
             data_format = params.get('data_format', 'base64')
             auth = cls(image, data, data_format,
                        storage_type) if data else None
+        elif enable_gluster:
+            if storage_type == 'glusterfs-direct':
+                peers = params.get("gluster_peers", "").split()
+                port = params.get("gluster_port")
+                transport = params.get("gluster_transport")
+                debug = params.get("gluster_debug")
+                logfile = params.get("gluster_logfile")
+                auth = cls(image, None, None, storage_type,
+                           debug=debug, logfile=logfile,
+                           transport=transport, port=port,
+                           peers=peers) if debug or logfile or peers else None
```

Comment:
Would you please elaborate on why it is not affected by `transport` or `port`, or `server` which is not set here?

Reply:
That's because transport/port/server can be retrieved from the uri itself(gluster+transport://server:port/), here we just need to pass the "additional" information for image access.